File: app/src/main/java/com/ericneidhardt/dynamicsoundboard/customview/EditTextBackEvent.java
package com.ericneidhardt.dynamicsoundboard.customview;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.AttributeSet;
import android.view.KeyEvent;
import android.widget.EditText;

/**
 * Project created by eric.neidhardt on 08.09.2014.
 */
public class EditTextBackEvent extends EditText
{
	private EditTextImeBackListener onImeBackListener;

	@SuppressWarnings("unused")
	public EditTextBackEvent(Context context)
	{
		super(context);
	}

	@SuppressWarnings("unused")
	public EditTextBackEvent(Context context, AttributeSet attrs)
	{
		super(context, attrs);
	}

	@SuppressWarnings("unused")
	public EditTextBackEvent(Context context, AttributeSet attrs, int defStyle)
	{
		super(context, attrs, defStyle);
	}

	public void setOnEditTextImeBackListener(EditTextImeBackListener listener)
	{
		this.onImeBackListener = listener;
	}

	@Override
	public boolean onKeyPreIme(int keyCode, @NonNull KeyEvent event)
	{
		if (event.getKeyCode() == KeyEvent.KEYCODE_BACK && event.getAction() == KeyEvent.ACTION_UP)
		{
			if (onImeBackListener != null)
				onImeBackListener.onImeBack(this, this.getText().toString());
		}
		return super.onKeyPreIme(keyCode, event);
	}

	public interface EditTextImeBackListener {
		public abstract void onImeBack(EditTextBackEvent ctrl, String text);
	}
}

File: android_utils/src/main/kotlin/org/neidhardt/androidutils/views/EditTextBackEvent.kt
package org.neidhardt.androidutils.views

import android.content.Context
import android.util.AttributeSet
import android.view.KeyEvent
import android.widget.EditText

/**
 * Project created by eric.neidhardt on 08.09.2014.
 */
class EditTextBackEvent : EditText {

	var onImeBackListener: EditTextImeBackListener? = null

	@SuppressWarnings("unused")
	constructor(context: Context) : super(context)

	@SuppressWarnings("unused")
	constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

	@SuppressWarnings("unused")
	constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle)

	override fun onKeyPreIme(keyCode: Int, event: KeyEvent): Boolean {
		if (event.keyCode == KeyEvent.KEYCODE_BACK && event.action == KeyEvent.ACTION_UP)
			this.onImeBackListener?.onImeBack(this, this.text.toString())

		return super.onKeyPreIme(keyCode, event)
	}

	interface EditTextImeBackListener {
		fun onImeBack(ctrl: EditTextBackEvent, text: String)
	}
}


